### DISCUSSION TITLE
How to maintain the aesthetics of the path?

### DISCUSSION THREAD

**Discussion by ZigglerPL** [community/community#107674] [Programming Help]
### Body

Hello, what causes the path element to have dimensions inconsistent with the content of the d attribute and to contain additional free space around itself that goes beyond the dimensions of the SVG element?

What are the key styles or other aspects that would lead the space in the path to aesthetically align with the rules specified in the d attribute?

`<div class="icon-container">

<svg viewBox="0 0 20 20">

<g>

<path d="M 1 0 L 19 9.5 L 1 19">

</g>

</svg>

</div>

.icon-container {

display: flex;

justify-content: center;

align-items: center;

width: 20px;

height: 20px;

}

SVG {

display: inline-block;

stroke: rgb(141, 145, 145);

stroke-width: 1.4px;

height: 100%;

width: 100%;

fill: none;

}`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by vatche-t** [comment]
The inconsistency in dimensions and additional space around a <path> element in SVG can be attributed to several factors, including the viewBox attribute of the SVG, the path's d attribute values, and CSS styles applied to the SVG or its container. Ensuring the viewBox matches the actual dimensions needed for the path, and adjusting the d attribute to correctly define the path's shape and size, are critical. CSS properties like width, height, and display should align with these dimensions to prevent extra space. Adjusting these aspects helps the path aesthetically align with its intended design.

**Comment by angelusrivera** [comment]
To make sure your path is getting aligned without all the weird craziness that comes with SVGs and CSS you might need to adjust the 'viewBox' just a little to match the bounds of your path. Also, def make sure the 'icon-container' has the right dimensions for the aspect ratio of ur SVG.

Say this is our SVG in our HTML:
```html
<div class="icon-container">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
        <path d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"/>
    </svg>
</div>
```

We need to make sure our view box is set up correctly as it is here. Also another thing, the stroke width could be messing with the size of it so definitely see if changing the values messes with the proportions! Have a good day <3

**Reply by ZigglerPL** [reply]
https://files.fm/f/nffnknradx
Thanks, but I don't understand something: why, for example, on Twitter, the visibility of the area in the SVG element is set to 0 0 24 24, and the size of the icon/path is e.g. 14x14 and yet it does not contain unnecessary empty space around it? So, apart from setting the visible area in the SVG element, there must be a reason why, if the icon has different dimensions, it does not contain empty spaces, as it does in the screenshot in the link.
